src/mixed.js

```diff
@@ -112,7 +112,7 @@ const proto = (SchemaType.prototype = {
         } and ${schema._type}`,
       );
     var cloned = this.clone();
-    var next = merge(this.clone(), schema.clone());
+    var next = prependDeep(schema.clone(), this.clone());
 
     // undefined isn't merged over, but is a valid value for default
     if (has(schema, '_default')) next._default = schema._default;
```

Comment:
this isn't true anymore

Reply:
Still true. If `this` has a non-`undefined` default value and `schema` has default value of `undefined`, then resulted one will have non-`undefined` one, we don't want that, we want to preserve override by `undefined`.

Reply:
I meant the comment. `undefined` isn't merged over from `this` unless it's also `undefined` in `schema`. the default value will still get clobbered yes. Tho i wonder if the `undefined` check in `prependDeep` should just be a `has` instead of strict undefined check

Reply:
Updated comment. This one will do? (If not, there is a feature in github where you can add a  suggestion here in comments and i can easily add it to PR)